### PR TITLE
Use standard dropdowns for combobox

### LINKED ISSUE #13: Combo box not accessible
Keyboard navigation doesn't seem to work at all on it

http://localhost:8080/jenkins/design-library/Select2/

(Moved to Textbox sample on rework branch)

@janfaracik did you have something to re-work this?

(can re-report this to core at some point, just wanted to keep track of issues in the current samples)

### PR DESCRIPTION
**Pull request by timja**
Fixes https://github.com/jenkinsci/design-library-plugin/issues/13

Unfortunately can't delete https://github.com/jenkinsci/jenkins/blob/master/war/src/main/webapp/scripts/combobox.js as `editableComboBox` still uses it, https://github.com/jenkinsci/jenkins/pull/9461 removes the last usage in core though.

This improves the usability and accessibility of the control by adding styling when doing keyboard navigation, and makes the code a lot simpler and easier to follow (last one used very old JavaScript design patterns and was hard to understand)

I haven't added support for `allowMultipleValues` as I couldn't find any usages of it.

### Testing done

Tested with the design library example http://localhost:8080/jenkins/design-library/Inputs/

Keyboard navigation, tab and enter on item
Clicking the item
Tabbing away before selecting an item
Editing items and making sure they show up

Unlike autocomplete, combobox shows items immediately so tested some special handling for that

I've also tested with https://github.com/jenkinsci/jenkins/pull/9461 (which was motivated by this PR)

### Proposed changelog entries

- Modernise combobox component

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

File: war/src/main/js/components/dropdowns/utils.js
Comment (timja) [inline war/src/main/js/components/dropdowns/utils.js:97]: occurs when navigating between potential text left and right without selecting an item (doesn't cause an issue just stacktraces in the console)

e.g. type North, then navigate left on the keyboard

File: war/src/main/js/components/combo-box/index.js
Comment (zbynek) [inline war/src/main/js/components/combo-box/index.js:8]: It might make sense to trigger form validation here, as done in #9453 (picking item from combobox may have fixed a previous validation error)

**Comment by zbynek** [comment]
Looks like quite a bit of code was copied from #9453, some parts were improved here (e.g. I missed the `focusout` listener). Maybe it would make sense to land this PR first, then refactor mine to avoid the duplication and make use of the improvements you've done here? Or would you prefer the other order?

**Comment by timja** [comment]
> Looks like quite a bit of code was copied from #9453, some parts were improved here (e.g. I missed the `focusout` listener). Maybe it would make sense to land this PR first, then refactor mine to avoid the duplication and make use of the improvements you've done here? Or would you prefer the other order?

I don't mind either way, I've just noticed that the Dynamic combobox example is broken though so will convert this back to draft and take another look later

**Comment by timja** [comment]
Retested all with the examples on http://localhost:8080/jenkins/design-library/Inputs/

I've refactored out some common code as well.

**Review by zbynek — APPROVED** [review]
Looks good!

**Comment by timja** [comment]
@janfaracik you able to check this please?

**Review by NotMyFault — APPROVED** [review]

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
